staging/src/k8s.io/client-go/rest/request.go

```diff
@@ -596,13 +596,16 @@ func (r *Request) tryThrottleWithInfo(ctx context.Context, retryInfo string) err
 		message = fmt.Sprintf("Waited for %v due to client-side throttling, not priority and fairness, request: %s:%s", latency, r.verb, r.URL().String())
 	}
 
-	if latency > longThrottleLatency {
-		klog.V(3).Info(message)
-	}
 	if latency > extraLongThrottleLatency {
-		// If the rate limiter latency is very high, the log message should be printed at a higher log level,
-		// but we use a throttled logger to prevent spamming.
+		// If the rate limiter latency is very high, the log message should be printed at a high log level, we use a
+		// throttled logger to prevent spamming.
 		globalThrottledLogger.Infof(message)
+	} else {
+		// even if extraLongThrottleLatency wasn't exceeded it's still possible longThrottleLatency was, log this with a
+		// lower level
+		if latency > longThrottleLatency {
+			klog.V(3).Info(message)
```

Comment:
the globalThrottledLogger is meant to be used to avoid spamming at higher log levels, for log level V(3), it is ok to log every request with latency > longThrottleLatency.

With this proposed change, the log messages may be throttled for requests with latency > extraLongThrottleLatency at log level V(3), which is not desirable.

Reply:
in all cases it will get logged
the current logic will log *twice* the same info mation as when t > 1000 ms it's also > 50ms so both trigger

this changes things so that it will only log once in this situation and log as it does now

Reply:
if latency > extraLongThrottleLatency, globalThrottledLogger.Infof(message) will be called, instead of klog.V(3).Info(message)

unlike klog.V(3).Info(message), which logs every message,  globalThrottledLogger.Infof(message) may throttle.